### ISSUE TITLE
BUG: in DataFrames(), parameters columns and index have some problems

### ISSUE THREAD

**Issue by LittleSec** [pandas-dev/pandas#20778]
```python
from scipy.io import netcdf
import numpy as np
ncFileName = 'sst_mask_cutlatlon.nc'
gridCSVName = 'mask_grid.csv'
f = netcdf.netcdf_file(ncFileName, 'r')
lon = f.variables['lon'].data # 251
lat = f.variables['lat'].data # 331
mask = f.variables['mask'].data[0] # (331, 251)
newMask = np.where(mask == 1, mask, 0)
pd.DataFrame(newMask, columns=lon, index=lat).to_csv(gridCSVName)
f.close()
```
#### Problem description

when I print(lat), the ouput is
```python
[21.225 21.275 21.325 21.375 21.425 ... 37.575 37.625 37.675 37.725] # len == 331
```
but if a print(pd.DataFrame(newMask, columns=lon, index=lat)), the index is:
```csv
21.2250004
21.2749996
21.3250008
21.375
21.4249992
21.4750004
21.5249996
21.5750008
21.625
...
```

why will this happen?

<details>

the nc file(89kb) is in this url: https://pan.baidu.com/s/1fJk6QYiETWlCj_bTO_xShQ

my environment is:
macOS Sierra and python 3.4.3

</details>

**Comment by TomAugspurger** [comment]
Can you try rewriting your issue as a minimal bug report? http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports It's not clear what the issue is or how I would reproduce it.

**Comment by LittleSec** [comment]
Thanks for your reply.
As you ask me to Craft Minimal Bug Reports, I tried to construct the data directly instead of reading the data from the file, to reproduce the problem, but directly constructing the data does not have this problem. So, it's not a bug.

Now, I think I have found where is the question.

I try to see their type:
```python
>>> print(type(lat[0]))
<class 'numpy.float32'>
>>> print(type(dt3.index[0]))
<class 'numpy.float64'>
```
**So, it is clear that the type conversions lead to precision problems.**

**In DataFrame() function, it will try to convert the type of parameters column and index  to float64, isn't it?**
